TypeCobol/Tools/Options-Config/TypeCobolConfiguration.cs

```diff
@@ -44,29 +140,183 @@ public static OptionSet GetCommonTypeCobolOptions(TypeCobolConfiguration typeCob
+                else if (config.InputFiles.Count != config.OutputFiles.Count)
+                    errorStack.Add(ReturnCode.OutputFileError, TypeCobolConfiguration.ErrorMessages[ReturnCode.OutputFileError]);
+            }
+            //SkeletonMissing
+            if (config.ExecToStep == ExecutionStep.Generate && config.skeletonPath.IsNullOrEmpty() && !errorStack.ContainsKey(ReturnCode.ExecToStepError))
+                errorStack.Add(ReturnCode.SkeletonMissing, TypeCobolConfiguration.ErrorMessages[ReturnCode.SkeletonMissing]);
+
+
+            //// Check unexpected token
+            if (unexpectedStrings.Count != 0)
+                errorStack.Add(ReturnCode.UnexpectedParamError, TypeCobolConfiguration.ErrorMessages[ReturnCode.UnexpectedParamError]);
+
+            //// Options values verification
+            //InputFileError
+            foreach (var path in config.InputFiles)
+            {
+                if (FileSystem.GetFiles(path, recursive: false).IsNullOrEmpty() && !errorStack.ContainsKey(ReturnCode.InputFileError))
+                {
+                    errorStack.Add(ReturnCode.InputFileError, TypeCobolConfiguration.ErrorMessages[ReturnCode.InputFileError]);
+                }
+            }
+            //SkeletonFileError
+            if (config.ExecToStep == ExecutionStep.Generate && !config.skeletonPath.IsNullOrEmpty() &&  !errorStack.ContainsKey(ReturnCode.ExecToStepError))
+            {
+                if (!File.Exists(config.skeletonPath))
+                {
+                    errorStack.Add(ReturnCode.SkeletonFileError, TypeCobolConfiguration.ErrorMessages[ReturnCode.SkeletonFileError]);
+                }
+            }
+            // EncodingError
+            switch (config.RawFormat.ToLower())
+            {
+                case "rdz":
+                    config.EncFormat = "rdz";
+                    config.Format = DocumentFormat.RDZReferenceFormat;
+                    break;
+                case "zos":
+                    config.EncFormat = "zos";
+                    config.Format = DocumentFormat.ZOsReferenceFormat;
+                    break;
+                case "utf8":
+                    config.EncFormat = "utf8";
+                    config.Format = DocumentFormat.FreeUTF8Format;
+                    break;
+                default:
+                    errorStack.Add(ReturnCode.EncodingError, TypeCobolConfiguration.ErrorMessages[ReturnCode.EncodingError]);
+                    break;
+            }
+            //IntrinsicError
+            foreach (var path in config.Copies)
+            {
+                if (FileSystem.GetFiles(path, recursive: false).IsNullOrEmpty() && !errorStack.ContainsKey(ReturnCode.IntrinsicError))
+                {
+                    errorStack.Add(ReturnCode.IntrinsicError, TypeCobolConfiguration.ErrorMessages[ReturnCode.IntrinsicError]);
+                }
+            }
+            //CopiesError
+            foreach (var path in config.CopyFolders)
```

Comment:
You can also refactor this loops that check files, they all do the same job.. the only diff is the returnCode, may you can create a method to do this job.